### PR TITLE
Get TSP files from remote git location with sparse checkout

### PR DESCRIPTION
**Pull request by gearama**
Step 1 in KV generation from TSP. 
download the specs localy from azure specs using git sparse checkout, thus only the folder we need vs the whole repo.
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: cmake-modules/TSPCompile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    execute_process(COMMAND ${GIT_EXECUTABLE} fetch
+         WORKING_DIRECTORY ${DOWNLOAD_FOLDER})
+    #switch branch
+    execute_process(COMMAND ${GIT_EXECUTABLE} checkout ${TSP_BRANCH}
+          WORKING_DIRECTORY ${DOWNLOAD_FOLDER})
+ 
+    if (NOT ${STATUS_CODE} EQUAL 0)
+        message(FATAL_ERROR "TSP download failed (Link: ${TSP_FULL_PATH}).")
+    endif()
+endmacro()
```
Comment (ahsonkhan) [inline cmake-modules/TSPCompile.cmake:39]: nit: new line at eof
```suggestion
endmacro()

```

File: cmake-modules/TSPCompile.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
+find_package(Git)
```
Comment (ahsonkhan) [inline cmake-modules/TSPCompile.cmake:1]: Missing license text

```suggestion
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.

find_package(Git)
```

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt [hunk 3]
```diff
@@ -31,6 +31,18 @@ include(AzureConfigRTTI)
...
 include(CreateMapFile)
+option(TSP_GEN "Generate from TypeSpec" OFF)
+
+if(TSP_GEN)
+    include(TSPCompile) 
+
+    set(TSP_DESTINATION "keyvault_secrets")
+    set(TSP_REPO "https://github.com/Azure/azure-rest-api-specs.git")
+    set(TSP_BRANCH "mccoyp/kv-secrets-tsp")
+    set(TSP_REPO_PATH "specification/keyvault/data-plane/Security.KeyVault.Secrets/")
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:42]: We are going to want to fetch from a sha. There will arise times where we want to build from a previous version of the tsp files.

File: cmake-modules/TSPCompile.cmake [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+macro(DownloadTSPFiles TSP_REPO TSP_BRANCH TSP_REPO_PATH TSP_DESTINATION)
+    message ("Downloading TSP files from ${TSP_REPO} branch ${TSP_BRANCH} path ${TSP_REPO_PATH} to ${TSP_DESTINATION}")
+
+    if(Git_FOUND)
+        message("Git found: ${GIT_EXECUTABLE}")
+    else()
+        message(FATAL_ERROR "Git not found")
+    endif()
```
Comment (RickWinter) [inline cmake-modules/TSPCompile.cmake:10]: Worth considering the developer experience. When an Azure service team adds the values to the library and gets them wrong. Will the tools fail in a way that makes sense and is easy to diagnose?

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt (hunk 3 above)
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:42]: Do we need repo and repo path? Can that be consolidated into one entry?

**Review by RickWinter — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt (hunk 3 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:42]: kinda, repo is the repo url ( https://..../xyz.git) the path is the subfolder, they are also diffrently used in commands, thus avoiding extra string parsing in cmake.

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:42]: true that

**Review by LarryOsterman — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt (hunk 3 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:42]: done

**Pull request merged**